app/addons/documents/mango/mango.components.react.jsx

```diff
@@ -121,35 +277,39 @@ function (app, FauxtonAPI, React, Stores, Actions, ReactComponents) {
         return;
       }
 
-      this.clearNotifications();
+      this.getMangoEditor().clearNotifications();
 
       Actions.saveQuery({
         database: this.state.database,
-        queryCode: this.refs.indexQueryEditor.getValue()
+        queryCode: this.getMangoEditor().getEditorValue()
       });
     }
   });
 
   var Views = {
-    renderHelpScreen: function (el) {
+    renderQueryEditor: function (el) {
       React.render(
-        <HelpScreen title={app.i18n.en_US['mango-help-title']} />,
+        <MangoQueryEditorController
+          description={app.i18n.en_US['mango-descripton']}
+          editorTitle={app.i18n.en_US['mango-title-editor']}
+          additionalIndexesText={app.i18n.en_US['mango-additional-indexes-heading']} />,
         el
       );
     },
-    removeHelpScreen: function (el) {
+    removeQueryEditor: function (el) {
       React.unmountComponentAtNode(el);
     },
     renderMangoIndexEditor: function (el) {
```

Comment:
Can you update all of these render functions to the new supported `setComponent` in the RouteObjects.